external-crates/move/move-compiler/src/parser/syntax.rs

```diff
@@ -1492,19 +1516,48 @@ fn parse_dot_or_index_chain(context: &mut Context) -> Result<Exp, Box<Diagnostic
         let exp = match context.tokens.peek() {
             Tok::Period => {
                 context.tokens.advance()?;
-                let n = parse_identifier(context)?;
-                if is_start_of_call_after_function_name(context, &n) {
-                    let call_start = context.tokens.start_loc();
-                    let mut tys = None;
-                    if context.tokens.peek() == Tok::Less && n.loc.end() as usize == call_start {
-                        let loc = make_loc(context.tokens.file_hash(), call_start, call_start);
-                        tys = parse_optional_type_args(context)
-                            .map_err(|diag| add_type_args_ambiguity_label(loc, diag))?;
+                let loc = current_token_loc(context.tokens);
+                match context.tokens.peek() {
+                    Tok::NumValue | Tok::NumTypedValue
+                        if context.env.check_feature(
+                            FeatureGate::PositionalFields,
+                            context.package_name,
+                            loc,
+                        ) =>
+                    {
+                        let contents = context.tokens.content();
+                        if parse_u8(contents).is_err() {
+                            let msg = format!(
+                                "Invalid field access. Expected a number less than or equal to {}",
+                                u8::MAX
+                            );
+                            let mut diag = diag!(Syntax::UnexpectedToken, (loc, msg));
+                            diag.add_note("Positional fields must be in the range [0 .. 255] and not be typed, e.g. `0_u8`");
+                            context.env.add_diag(diag);
+                        };
+                        context.tokens.advance()?;
+                        let field_access = Name::new(loc, contents.into());
+                        Exp_::Dot(Box::new(lhs), field_access)
+                    }
+                    _ => {
```

Comment:
Could do `matches` above if you want to unindent one level. This is fine too though :)